src/Servers/IIS/src/AspNetCoreModuleV2/CommonLib/RegistryKey.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "RegistryKey.h"
+#include "exceptions.h"
+
+std::optional<DWORD> RegistryKey::TryGetDWORD(HKEY section, const std::wstring& subSectionName, const std::wstring& valueName, DWORD flags)
+{
+    DWORD dwData = 0;
+    DWORD cbData = sizeof(dwData);
+    if (LOG_LAST_ERROR_IF(RegGetValue(section, subSectionName.c_str(), valueName.c_str(), RRF_RT_REG_DWORD | flags, nullptr, reinterpret_cast<LPBYTE>(&dwData), &cbData) != NO_ERROR))
+    {
+        return std::nullopt;
+    }
+
+    return dwData;
+}
+
+std::optional<std::wstring> RegistryKey::TryGetString(HKEY section, const std::wstring& subSectionName, const std::wstring& valueName, DWORD flags)
+{
+    DWORD cbData;
+
+    if (LOG_LAST_ERROR_IF(RegGetValue(section, subSectionName.c_str(), valueName.c_str(), RRF_RT_REG_SZ | flags, nullptr, nullptr, &cbData) != NO_ERROR))
```

Comment:
Consolidate where we are setting flags; either remove flags as a parameter and put them all in the call to RegGetValue or remove `RRF_RT_REG_SZ |` and let the caller set.

Reply:
Some flags are responsibilities of the caller and some are responsibilities of the callee.
In this case, the calee enforces that string is expected but the caller is free to change other flags like which hive we are looking at.

Reply:
I'll buy that argument.